pymatviz/ptable.py

```diff
@@ -699,292 +730,438 @@ def histogram(
         ax.spines[["right", "top", "left"]].set_visible(False)
 
 
-def ptable_heatmap(
-    values: ElemValues,
-    *,
-    log: bool | Normalize = False,
-    ax: plt.Axes | None = None,
-    count_mode: ElemCountMode = ElemCountMode.composition,
```

Comment:
I'm a bit unsure about `count_mode` at this moment. I think for our current implementation, the `data` is expected to be an element to value mapping.

It looks like this (`count_elements`) is a preprocessing function, perhaps we should separate the preprocessing stage from the plotter?